### PR TITLE
block rejected validation receipts 4eva

### PR DESCRIPTION
**Pull request by thedavidmeister**
### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Review by neonphog — APPROVED** [review]
:guitar:

File: crates/holochain/src/core/workflow/validation_receipt_workflow.rs [hunk 1]
```diff
@@ -99,6 +101,17 @@ pub async fn validation_receipt_workflow(
             continue;
         }
 
+        if matches!(receipt.validation_status, ValidationStatus::Rejected) {
+            // Block BEFORE we integrate the outcome because this is not atomic
```
Comment (jost-s) [inline crates/holochain/src/core/workflow/validation_receipt_workflow.rs:105]: 💡 👯

**Review by jost-s — APPROVED** [review]
Yeah, would be great to have a test as you said.

**Comment by steveeJ** [comment]
@thedavidmeister cachix setup has been removed on develop. if you pull those changes into your base branch the checks should be fine. let me know if you need to cherry-pick the exact changes and i can help you out.

File: crates/holochain/src/core/workflow/app_validation_workflow.rs [hunk 2]
```diff
@@ -206,7 +205,7 @@ async fn app_validation_workflow_inner(
                         }
                         Outcome::Rejected(_) => {
                             rejected += 1;
-                            tracing::warn!("Received invalid op! Warrants aren't implemented yet, so we can't do anything about this right now, but be warned that somebody on the network has maliciously hacked their node.\nOp: {:?}", op_light);
+                            tracing::warn!("Received invalid op. The op author will be blocked.\nOp: {:?}", op_light);
```
Comment (maackle) [inline crates/holochain/src/core/workflow/app_validation_workflow.rs:208]:
```suggestion
                            tracing::info!("Received invalid op. The op author will be blocked.\nOp: {:?}", op_light);
```

Feels more like info than a warning

**Review by maackle — APPROVED** [review]
Hard to believe this is all it takes to have an immune system :sweat_smile:

**Pull request merged**